azure-pipelines.yml

```diff
@@ -1,342 +1,421 @@
+                  terraformSubscriptionID: ${{ component.terraformSubscriptionID }}
+                  backendAzureRmResourceGroupName: ${{ component.backendAzureRmResourceGroupName }}
+                  setClusterNumbersEnvVar: true
+                  ${{ if ne(parameters['cluster'], 'All') }}:
+                    targetCommand: >
+                      -target azurerm_resource_group.kubernetes_resource_group["\"${{parameters.cluster}}\""]
+                      -target module.kubernetes["\"${{parameters.cluster}}\""]
 
-  - stage: PipelineTests
-    displayName: "PipelineTests"
-    dependsOn: BootStrapClusters
-    jobs:
-      - job: PipelineTests
-        pool:
-          vmImage: ${{ variables.agentPool }}
-        timeoutInMinutes: ${{ variables.timeoutInMinutes }}
+          - job: "DeployInfrastructuremis"
+            dependsOn: DeployInfrastructure
+            steps:
+              - template: pipeline-steps/deploy-service.yaml
+                parameters:
+                  environment: ${{ component.env }}
+                  location: ${{ parameters.location }}
+                  stack: "aks-mis"
+                  project: $(project)
+                  tfInitSub: ${{ variables.tfInitSub }}
+                  serviceConnection: ${{ component.serviceConnection }}
+                  action: ${{ parameters.action }}
+                  setClusterNumbersEnvVar: false
+                  terraformSubscriptionID: ${{ component.terraformSubscriptionID }}
+                  backendAzureRmResourceGroupName: ${{ component.backendAzureRmResourceGroupName }}
+                  ${{ if ne(parameters['cluster'], 'All') }}:
+                    targetCommand: >
+                      -target azurerm_role_assignment.uami_cft_rg_identity_operator["\"${{parameters.cluster}}\""]
+                      -target azurerm_role_assignment.preview_externaldns_read_rg["\"${{parameters.cluster}}\""]
+                      -target azurerm_role_assignment.preview_externaldns_dns_zone_contributor["\"${{parameters.cluster}}\""]
+                      -target azurerm_role_assignment.preview2aat_cft_rg_identity_operator["\"${{parameters.cluster}}\""]
+                      -target data.azurerm_kubernetes_cluster.kubernetes["\"${{parameters.cluster}}\""]
+
+
+      - stage: "BootStrapClusters_${{ component.env }}"
+        displayName: "${{ upper(component.env) }}: BootStrap Clusters"
+        dependsOn:
+          - "Aks_${{ component.env }}"
         condition: |
           or(
-            and(succeeded(), eq('${{ parameters.action }}', 'apply')),
-            and(succeeded(), eq(variables['isMain'], true), eq(variables['isAutoTriggered'], true))
+            and(eq('${{ parameters.cluster }}', 'All'), eq(variables['Build.Reason'], 'Manual')),
+            and(in(dependencies.CheckingClusters_${{ component.env }}.outputs['CheckingClusters.check_clusters.cluster_deploy'], 'All', '00', '01'), eq(variables['Build.Reason'], 'Manual')),
```

Comment:
What is the point in this condition? The cluster will only ever be one of those values if you run it manually? Similar with the ones above and below it, what are they trying to stop happening?